app/Http/Controllers/V2/Dashboard/CountriesController.php

```diff
@@ -20,22 +21,40 @@ public function __invoke(): CountriesResource
 
     public function getAllCountries()
     {
+        $projectsCountrieslug = Project::where('framework_key', 'terrafund')
+            ->whereHas('organisation', function ($query) {
+                $query->whereIn('type', ['for-profit-organization', 'non-profit-organization']);
+            })->pluck('country');
         $countryId = FormOptionList::where('key', 'countries')->value('id');
         $countries = FormOptionListOption::where('form_option_list_id', $countryId)
             ->orderBy('label')
+            ->select('id', 'label', 'slug')
             ->get();
         $countriesResponse = [];
         foreach ($countries as $country) {
-            $countriesResponse[] = [
-                'country_slug' => $country->slug,
-                'id' => $country->id,
-                'data' => (object) [
-                    'label' => $country->label,
-                    'icon' => '/flags/' . $country->label . '.svg',
-                ],
-            ];
+            if ($this->hasProjectsInCountry($country->slug, $projectsCountrieslug)) {
+                $countriesResponse[] = [
+                    'country_slug' => $country->slug,
+                    'id' => $country->id,
+                    'data' => (object) [
+                        'label' => $country->label,
+                        'icon' => '/flags/' . $country->label . '.svg',
+                    ],
+                ];
+            }
         }
 
         return $countriesResponse;
     }
+
+    public function hasProjectsInCountry($country, $projectsSlug)
+    {
+        $projects = $projectsSlug->contains($country);
+        if ($projects) {
```

Comment:
rename this variable to `hasProjects` or something similar.

and you can return this directly `return $hasProjects`